src/charm.py

```diff
@@ -91,6 +90,11 @@ def _mongodb_client_relation_changed(self, event: ops.framework.EventBase) -> No
         )
         if initial != self._stored.reldata["mongodb"]:
             self._configure_workload(event)
+    
+    def _check_external_hostname(self, hostname):
```

Comment:
The method name seems a bit ambiguous. I wouldn't expect it to return a value. Can you rename it?